public/src/modules/uploader.js

```diff
@@ -107,9 +116,11 @@ define('uploader', ['csrf', 'translator'], function(csrf, translator) {
 		return response;
 	}
 
-	module.hideAlerts = function(modal) {
-		$(modal).find('#alert-status, #alert-success, #alert-error, #upload-progress-box').addClass('hide');
-	};
+	function hasValidFileSize(fileElement, maxSize) {
+		if (window.FileReader && maxSize) {
+			return fileElement.files[0].size <= maxSize * 1000;
+		}
```

Comment:
Should be ok now.

I'm thinking about implementing the same behavior for the composer plugins. composer-default doesn't use the upload-modal. What's your preference there? Shall the current way (simple open-file dialog) remain or would you be ok with using the modal there, too?
I'd use the modal, for UX consistency.